ami/trainers/components/auditory.py

```diff
@@ -0,0 +1,89 @@
+from pathlib import Path
+
+import torch
+import torch.nn as nn
+import torchaudio
+from torch.utils.data import Dataset
+
+
+class IntervalSamplingAudioDataset(Dataset[tuple[torch.Tensor]]):
+    """Dataset class for sampling audios at regular intervals from a directory.
+
+    This class creates a dataset by sampling audios at fixed intervals from a specified directory.
+    It supports .wav formats and can optionally pre-load audios into memory for faster access.
+
+    Key Features:
+    - Samples audios at regular intervals from a directory.
+    - Supports .wav format.
+    - Applies specified transformations to the audios.
+    - Option to pre-load audios into memory for faster access.
+
+    Usage:
+    - Initialize the dataset with the desired parameters.
+    - Use it with PyTorch DataLoader for efficient batch processing.
+
+    Example:
+        dataset = IntervalSamplingAudioDataset("/path/to/audios", sample_rate=16000, num_select=1000)
+        dataloader = DataLoader(dataset, batch_size=32, shuffle=True)
+
+    Note:
+    - Pre-loading audios can significantly increase memory usage for large datasets.
+    """
+
+    def __init__(
+        self,
+        audio_dir: str | Path,
+        sample_rate: int,
+        num_select: int,
+        pre_loading: bool = True,
```

Comment:
音声データの長さを定められる `sample_size`がほしい
```suggestion
        sample_size: int,
        pre_loading: bool = True,
```

Reply:
https://github.com/MLShukai/ami/pull/279/commits/e4ccec12284854be5c58cf99a75ff9ae2bbde503 で追加